src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -68,145 +72,168 @@ public Condition getSqlCondition(String fqlCriteria, EntityType entityType) {
+  private static Condition handleAnd(AndCondition andCondition, EntityType entityType, Field<Object> field) {
     return and(andCondition.value().stream().map(c -> getSqlCondition(c, entityType)).toList());
   }
 
-  private Condition handleRegEx(RegexCondition regExCondition, EntityType entityType) {
+  private static Condition handleRegEx(RegexCondition regexCondition, EntityType entityType, Field<Object> field) {
     // perform case-insensitive regex search
-    return condition("{0} ~* {1}", field(regExCondition, entityType), val(regExCondition.value()));
+    return condition("{0} ~* {1}", field, val(regexCondition.value()));
   }
 
-  private Condition handleContains(ContainsCondition containsCondition, EntityType entityType) {
-    if (containsCondition.value() instanceof String s) {
-      return field(containsCondition, entityType).containsIgnoreCase(s);
+  private static Condition handleContains(ContainsCondition containsCondition, EntityType entityType, Field<Object> field) {
+    if (containsCondition.value() instanceof String value) {
+      return caseInsensitiveComparison(containsCondition, entityType, field, value, Field::containsIgnoreCase, Field::contains);
     }
-    return field(containsCondition, entityType).contains(containsCondition.value());
+    return field.contains(containsCondition.value());
   }
 
-  private Condition handleNotContains(NotContainsCondition notContainsCondition, EntityType entityType) {
-    Field<Object> field = field(notContainsCondition, entityType);
-    if (notContainsCondition.value() instanceof String s) {
-      return field.notContainsIgnoreCase(s).or(field(notContainsCondition, entityType).isNull());
+  private static Condition handleNotContains(NotContainsCondition notContainsCondition, EntityType entityType, Field<Object> field) {
+    if (notContainsCondition.value() instanceof String value) {
+      return caseInsensitiveComparison(notContainsCondition, entityType, field, value, Field::notContainsIgnoreCase, Field::notContains)
+        .or(field.isNull());
     }
-    return field.notContains(notContainsCondition.value()).or(field(notContainsCondition, entityType).isNull());
+    return field.notContains(notContainsCondition.value()).or(field.isNull());
+  }
+
+  /**
+   * Create a case-insensitive comparison of the given fieldCondition with the given entityType.
+   * <p /><b>Note: This method always converts the comparison value to lower-case, while the column will be converted
+   * to lower-case only if it does not have a filterValueGetter (currently, we assume the filterValueGetter
+   * will convert the column to lower-case. If that assumption changes, this method will need to be updated)</b>
+   *
+   * @param toCaseInsensitiveCondition The function to build a Condition where the column is converted to lower-case
+   * @param toCaseSensitiveCondition   The function to build a Condition where the column is *not* converted to lower-case
```

Comment:
I only bothered documenting these 2 params, since the others seemed pretty obvious (they are the same as in pretty much every other method in the class)

Reply:
I supposed `value` isn't used everywhere, so it's not as obvious. I'll add something for that one, too